app/Helpers/database/player-achievement.php

```diff
@@ -485,22 +492,52 @@ function getAchievementDistribution(
     }
 
     // Returns an array of the number of players who have achieved each total, up to the max.
-    $query = "
+    if (env('FEATURE_AGGREGATE_QUERIES')) {
+        if ($flag === AchievementFlag::OfficialCore) {
+            $countColumn = $hardcore ? 'achievements_unlocked_hardcore' : 'achievements_unlocked';
+            $query = "SELECT pg.$countColumn AS AwardedCount, COUNT(*) AS NumUniquePlayers
+                      FROM player_games AS pg
+                      $joinStatementNew
+                      WHERE pg.game_id = :gameId AND pg.$countColumn > 0
+                      $requestedByStatement
+                      GROUP BY AwardedCount
+                      ORDER BY AwardedCount DESC";
+            unset($bindings['achievementFlag']);
```

Comment:
**game/1446**
With feature: 121ms
Without feature: 115ms

Reply:
Do you see a need for improvement here, like an index that might improve group by performance? (Not sure if that'd work actually)
Or are we happy with the results so far?

Reply:
I'm not sure that an additional index would help. The query boils down to:
```
SELECT pg.achievements_unlocked AS AwardedCount, COUNT(*) AS NumUniquePlayers
FROM player_games AS pg
WHERE pg.game_id = 1446 AND pg.achievements_unlocked > 0
GROUP BY AwardedCount
ORDER BY AwardedCount DESC
```

And there's already an index on game_id.

It's just another aggregation issue. The time is most likely spent tallying the 30k records into 150 buckets. The query is repeated again for softcore, and the results of both queries are used to generate the achievement distribution graph. I think the best solution is to move that to a secondary page for the game.

Reply:
Sounds good - let's have that as soon as we work on those subpages.